ff/src/fields/mod.rs

```diff
@@ -209,6 +209,94 @@ pub trait Field:
     }
 }
 
+/// Fields that have a cyclotomic subgroup, and which can leverage efficient inversion
+/// and squaring operations for elements in this subgroup.
+pub trait CyclotomicField: Field {
```

Comment:
New field that should be implemented for fields that have more efficient operations for their cyclotomic subgroup, such as faster inverses or squaring.